### ISSUE TITLE
Specific support for file:/// protocol

### ISSUE THREAD

**Issue by merkys** [EnjoyCpp/jsonapi-client-cpp#33]
[file:/// protocol](https://en.wikipedia.org/wiki/File_URI_scheme) is used to refer to local files in URIs. It would be beneficial (mostly for testing) to implement specific support for it. Already, `curl` is able to get the contents of a file:
```bash
andrius@ondatra ~ $ curl file:///etc/hostname
ondatra
```
Thus, if one would create a `Server` with URL given as `file:///usr/share/my-database`, code could detect used `file:///` protocol and do the following:

* `get_one()`: return the contents of a file (`file:///usr/share/my-database/<type>/<id>`)
* `get_all()`: return the contents of a directory (`file:///usr/share/my-database/<type>`) in a single JSON response (let's assume all files in the directory are JSON files)
* create/update record: save JSON to a file (`file:///usr/share/my-database/<type>/<id>`)
* delete record: delete a file (`file:///usr/share/my-database/<type>/<id>`)

This feature has to wait for #27 to be completed first.

**Comment by EnjoyCpp** [comment]
@merkys I don't really have ideas how to work with  `file:/// protocol` in code. Maybe there is an some kind of example?

**Comment by merkys** [comment]
Requests with `file:///` protocol should be handled without using `curl`, but simply by operating with file system. That is, for example, `Server::get_one(type, id)` should check its URL, and if it starts with `file://`, it should read a local file instead of using `curl`.

For example, I create a JSON file and store it in `/home/andrius/json/my-file.json`. The following should get me its contents:
```c++
Server home( "file:///home/andrius" ); // my server actually points to '/home/andrius' in my file system
Resource my_file = home.get_one( "json", "my-file.json" ); // translates to file:///home/andrius/json/my-file.json
```

**Comment by EnjoyCpp** [comment]
@merkys So i use this to get where my server points:
```bash
enjoy@Enjoy-Lenovo-Z51-70:~$ curl file:///etc/hostname
Enjoy-Lenovo-Z51-70
```
And then i modify `get_one()` and `get_all()` to get URL from files in my file system rather than from website?

**Comment by merkys** [comment]
> @merkys So i use this to get where my server points:
> 
> ```shell
> enjoy@Enjoy-Lenovo-Z51-70:~$ curl file:///etc/hostname
> Enjoy-Lenovo-Z51-70
> ```
> 
> And then i modify `get_one()` and `get_all()` to get URL from files in my file system rather than from website?

Not exactly. `/etc/hostname` happens to be a simple file containing your computer name. By using `curl file://<any path>` you may get contents of any file on your computer. To simulate `curl file:///etc/hostname` you should be able to do the following:
```c++
Server home( "file://" );
Resource my_file = home.get_one( "etc", "hostname" ); // translates to file:///etc/hostname
```

**Edit**: somehow this code does not work now - need to investigate why.

**Comment by EnjoyCpp** [comment]
@merkys i have to do something similar to: 
`https://github.com/EnjoyCpp/jsonapi-client-cpp/blob/5d648ec3c2ebbf2ccd1cb84b11fd69ec7008c983/src/Server.cpp#L92`

**Comment by EnjoyCpp** [comment]
@merkys i have to do something similar to?:
https://github.com/EnjoyCpp/jsonapi-client-cpp/blob/5d648ec3c2ebbf2ccd1cb84b11fd69ec7008c983/src/Server.cpp#L92-L95

**Comment by merkys** [comment]
> @merkys i have to do something similar to?:
> 
> https://github.com/EnjoyCpp/jsonapi-client-cpp/blob/5d648ec3c2ebbf2ccd1cb84b11fd69ec7008c983/src/Server.cpp#L92-L95

Yes, this should fetch the content of file located at `/home/enjoy/JSON/my-file.json`.

It would be great to open a separate git branch for testing/implementing this feature. Testing and developing in `master` will most likely break tests.

**Comment by EnjoyCpp** [comment]
@merkys Somehow method get_one fails :
`make: *** [test] Segmentation fault (core dumped)`
https://github.com/EnjoyCpp/jsonapi-client-cpp/blob/5806d6e70532797cb286961f01130d5896081dfe/src/Server.cpp#L99-L102

Also is this way for get_all would be correct?
https://github.com/EnjoyCpp/jsonapi-client-cpp/blob/5806d6e70532797cb286961f01130d5896081dfe/src/Server.cpp#L81-L84
